src/test/java/com/amazon/ion/impl/EncodingDirectiveCompilationTest.java

```diff
@@ -51,10 +52,12 @@ public class EncodingDirectiveCompilationTest {
 
     private static final int FIRST_LOCAL_SYMBOL_ID = 1;
 
-    private static void assertMacroTablesEqual(IonReader reader, StreamType streamType, SortedMap<String, Macro> expected) {
+    private static void assertMacroTablesContainsExpectedMappings(IonReader reader, StreamType streamType, SortedMap<String, Macro> expected) {
         Map<MacroRef, Macro> expectedByRef = streamType.newMacroTableByMacroRef(expected);
-        Map<MacroRef, Macro> actual = streamType.getEncodingContext(reader).getMacroTable();
-        assertEquals(expectedByRef, actual);
+
+        MacroTable actual = streamType.getEncodingContext(reader).getMacroTable();
+        // TODO: This assertion is weak, we don't know that the actual macro table contains *only* the expectations
+        expectedByRef.forEach((k,v) -> assertEquals(v, actual.get(k)));
```

Comment:
This may be a good thing to fix before we forget. Would just adding the following be sufficient?
```
// Assuming addition of `asMap()` that returns Map<MacroRef, Macro>, or something similar
actual.asMap().forEach((k, v) -> assertEquals(v, expectedByRef.get(k)));
```

Reply:
At present `MacroTable` isn't enumerable. I could fix that fairly easily in MutableMacroTable if I create a method in SystemMacro... 🤔